src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,85 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final int AMOUNT_INDEX = 1;
+    public static final int OPERATION_TYPE_INDEX = 0;
+    public static final int SUPPLY_DATA_INDEX = 0;
+    public static final int VALUE_FROM_DATA_INDEX = 1;
+    public static final int BUY_DATA_INDEX = 1;
+    public static final int SUPPLY_DATA_IN_REPORT_INDEX = 0;
+    public static final int BUY_DATA_IN_REPORT_INDEX = 1;
+    public static final int RESULT_DATA_IN_REPORT_INDEX = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeReport(toFileName, report);
+    }
+
+    private String readFile(String fromFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line = reader.readLine();
+            while (line != null) {
+                String[] data = line.split(",");
+                String operationType = data[OPERATION_TYPE_INDEX].trim();
+                int amount = Integer.parseInt(data[AMOUNT_INDEX].trim());
+
+                if ("supply".equals(operationType)) {
+                    totalSupply += amount;
+                } else if ("buy".equals(operationType)) {
+                    totalBuy += amount;
+                }
+
+                line = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot read file", e);
+        }
+
+        return "Total Supply:" + totalSupply + ";Total Buy:" + totalBuy;
```

Comment:
create int result = 0; calculate it result = supplyResult - buyResult; and 
return new int[] {supplyResult, buyResult, result};

Reply:
I agree with the approach, but i think it is better to remove here only `int[] {supplyResult, buyResult}`;
I think that result should be calculated in createReport method to not break SRP